lib/matplotlib/ticker.py

```diff
@@ -377,12 +381,32 @@ class FuncFormatter(Formatter):
     """
     Use a user-defined function for formatting.
 
-    The function should take in two inputs (a tick value ``x`` and a
-    position ``pos``), and return a string containing the corresponding
-    tick label.
+    The function can take in at most two inputs (a required tick value ``x``
+     and an optional position ``pos``), and must return a string containing
+     the corresponding tick label.
     """
     def __init__(self, func):
-        self.func = func
+        if not isinstance(func, types.BuiltinFunctionType):
+            nargs = len(inspect.signature(func).parameters)
+        elif (func.__name__ == 'format'):
+            #check if there's a format spec
+            nargs = len([(_, _, fs, _) for (_, _, fs, _)
+                         in string.Formatter().parse(func.__self__)
+                         if fs is not None])
+        else:
+            #finding argcount for other builtins is a mess
+            nargs = 2
+            raise warnings.warn(f"{func.__name__} is not supported "
+                                 "and may not work as expected")
+
+        if nargs == 1:
+            self.func = lambda x, pos: func(x)
```

Comment:
that won't be picklable; you need to store nargs as a private attribute and switch over it in `__call__`.

Reply:
will do but slightly confused why - are there more tests I should be writing?

Reply:
depends on how much work you want to do, heh.

Reply:
🤷 if it's important will do, but would like to know why.